test/controllers/votes_controller_test.rb

```diff
@@ -0,0 +1,34 @@
+require "test_helper"
+
+describe VotesController do
+  it "should get new" do
+    get votes_new_url
+    value(response).must_be :success?
+  end
+
+  it "should get destroy" do
+    get votes_destroy_url
+    value(response).must_be :success?
+  end
+
+  it "should get upvote" do
+    get votes_upvote_url
+    value(response).must_be :success?
+  end
+
+  it "should get index" do
+    get votes_index_url
+    value(response).must_be :success?
+  end
+
+  it "should get show" do
+    get votes_show_url
+    value(response).must_be :success?
+  end
+
+  it "should get create" do
+    get votes_create_url
+    value(response).must_be :success?
+  end
+
```

Comment:
You should be testing the uniqueness constraint here! Specific test cases I'd want to see:

- A user can have votes for two different works
- A work can have votes from two different users
- A user cannot vote for the same work twice